src/parser.js

```diff
@@ -0,0 +1,51 @@
+// This is the parser to handle formatting messages to PS 
+// or GitHub, depending on which is  called
+
+'use strict'
+
+var escape = require('escape-html')
+
+exports.getRepoName = function (repo) {
+    switch (repo) {
+        case 'Pokemon-Showdown':
+            return 'server'
+        case 'Pokemon-Showdown-Client':
+            return 'client'
+        case 'Pokemon-Showdown-Dex':
+            return 'dex'
+        default:
+            return repo.toLowerCase()
+      }
+}
+
+exports.formatPR = function (location, repo, author, action, num, title, url) {
```

Comment:
it looks like you want a `Formatter` interface with `formatPR` and `formatPush` methods and then `DiscordFormatter` and `ShowdownFormatter` implementations?

Reply:
Are interfaces TypeScript? I was hoping to avoid TypeScript for this initial implementation